src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowTransformerControlTest.java

```diff
@@ -0,0 +1,442 @@
+        assertEquals(2, t2wt.getRatioTapChanger().getTapPosition());
+    }
+
+    @Test
+    void remoteVoltageControlT2wtTest() {
+        selectNetwork(createNetworkWithT2wt());
+
+        parameters.setTransformerVoltageControlOn(true);
+        t2wt.getRatioTapChanger()
+                .setTargetDeadband(0)
+                .setRegulating(true)
+                .setTapPosition(0)
+                .setRegulationTerminal(t2wt.getTerminal1())
+                .setTargetV(134.0);
+
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        assertVoltageEquals(135.0, bus1);
+        assertVoltageEquals(132.640, bus2); //FIXME: should be 134.281
+        assertVoltageEquals(3.705, bus3); //FIXME: should be 34.433
+        assertTrue(result.isOk());
+        assertEquals(0, t2wt.getRatioTapChanger().getTapPosition());
+    }
+
+    @Test
+    void baseCaseT3wtTest() {
+        selectNetwork(createNetworkWithT3wt());
+
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        assertTrue(result.isOk());
+
+        assertVoltageEquals(135.0, bus1);
+        assertVoltageEquals(134.265, bus2);
+        assertVoltageEquals(34.402, bus3);
+        assertVoltageEquals(10.320, bus4);
+    }
+
+    @Test
+    void tapPlusTwoT3wtTest() {
+        selectNetwork(createNetworkWithT3wt());
+
+        t3wt.getLeg2().getRatioTapChanger().setTapPosition(2);
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        assertTrue(result.isOk());
+
+        assertVoltageEquals(135.0, bus1);
+        assertVoltageEquals(134.264, bus2);
+        assertVoltageEquals(28.147, bus3);
+        assertVoltageEquals(10.320, bus4);
+    }
+
+    @Test
+    void voltageControlT3wtTest() {
+        selectNetwork(createNetworkWithT3wt());
+
+        t3wt.getLeg2().getRatioTapChanger()
+                .setTargetDeadband(0)
+                .setRegulating(true)
+                .setTapPosition(0)
+                .setRegulationTerminal(t3wt.getLeg2().getTerminal())
+                .setTargetV(28.);
+
+        parameters.setTransformerVoltageControlOn(true);
+
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        assertTrue(result.isOk());
+        assertVoltageEquals(28.147, bus3);
+        assertEquals(2, t3wt.getLeg2().getRatioTapChanger().getTapPosition());
+    }
+
+    @Test
+    void remoteVoltageControlT3wtTest() {
+        selectNetwork(createNetworkWithT3wt());
+
+        t3wt.getLeg2().getRatioTapChanger()
+                .setTargetDeadband(0)
+                .setRegulating(true)
+                .setTapPosition(0)
+                .setRegulationTerminal(t3wt.getLeg3().getTerminal())
+                .setTargetV(10.0);
+
+        parameters.setTransformerVoltageControlOn(true);
+
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        assertTrue(!result.isOk()); //FIXME
```

Comment:
@geofjamg something is not working at all because the solver failed in that case. The RTC is located on leg2 and I am trying to regulate voltage on leg3. Note that the bus1 is PV through a generator. I am not sure to get what is happening.